jena-tdb/src/main/java/org/apache/jena/tdb/base/block/BlockMgrCache.java

```diff
@@ -58,10 +58,7 @@ private BlockMgrCache(int readSlots, int writeSlots, final BlockMgr blockMgr)
     {
```

Comment:
I've made changes to `BlockMgrCache` sufficient to permit the removal of `Cache0` from the codebase, but I'd much appreciate a careful review here. Also, I have to admit I think this new code (with all the `null`-checks) to be pretty ugly. I'd like to rewrite it using `Optional`, but I'd first like to be sure that my basic understanding is correct.

P.S. Actually, I'd like to make `Cache::getIfPresent` return `Optional`. That would really eliminate a lot of `null`-checking.